### PR TITLE
Add support for $type aliasses.

### PR DESCRIPTION
**Pull request by sebakerckhof**
Fix: https://github.com/meteor/meteor/issues/9401

Note: I added some extra code that checks the operand values which is not strictly necessary, but now the errors are exactly the same as when using mongo shell.

### CODE REVIEW

File: packages/minimongo/common.js [hunk 1]
```diff
@@ -87,8 +87,41 @@ export const ELEMENT_OPERATORS = {
...
+          'dbPointer': 12,
+          'javascript': 13,
+          'symbol': 14,
+          'javascriptWithScope': 15,
+          'int': 16,
+          'timestamp': 17,
+          'long': 18,
+          'decimal': 19,
+          'minKey': -1,
+          'maxKey': 127,
```
Comment (benjamn) [inline packages/minimongo/common.js]: Is there any way we could get this information from some npm package, or from the `mongo` package, somehow?

File: packages/minimongo/common.js [hunk 2]
```diff
@@ -87,8 +87,41 @@ export const ELEMENT_OPERATORS = {
...
+        };
+        if (!operandAliasMap.hasOwnProperty(operand)) {
```
Comment (abernix) [inline packages/minimongo/common.js]: In order to protect against the (unlikely, but seemingly more likely than usual case, given that this is a list of operands) scenario where operands includes a property called `hasOwnProperty`, could we switch this to?

```js
Object.prototype.hasOwnProperty.call(operandAliasMap, operand);
```

File: packages/minimongo/common.js (hunk 1 above)
Comment (sebakerckhof) [reply packages/minimongo/common.js]: Did a quick search and couldn't find anything (the mongodb npm driver at least doesn't contain it). Also, the `mongo` package has `minimongo` as a dependency, not the other way around.

File: packages/minimongo/common.js (hunk 2 above)
Comment (sebakerckhof) [reply packages/minimongo/common.js]: yep, apparently there was already a `hasOwn` shorthand foreseen.

Comment (abernix) [reply packages/minimongo/common.js]: Yeah, we frequently opt to use that technique! Always feel free to use it as you see fit! :smile:

File: packages/minimongo/common.js (hunk 1 above)
Comment (benjamn) [reply packages/minimongo/common.js]: OK, I can't find anything useful either!

**Pull request merged**